File: src/interface/are-files-less-than-request.ts
export default class AreFilesLessThanRequest {
  rootPath: string
  num: number
  noSelectedPaths: string[]
  excludeDirs: string[]
  excludeExts: string[]
  enableGitignore: boolean
  
  constructor(rootPath: string, num: number, noSelectedPaths: string[], excludeDirs: string[], excludeExts: string[], enableGitignore: boolean) {
    this.rootPath = rootPath
    this.num = num
    this.noSelectedPaths = noSelectedPaths
    this.excludeDirs = excludeDirs
    this.excludeExts = excludeExts
    this.enableGitignore = enableGitignore
  }
}

File: src-tauri/src/request/are_files_less_than_request.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AreFilesLessThanRequest {
    pub root_path: String,
    pub num: i32,
    pub no_selected_paths: Vec<String>,
    pub exclude_dirs: Vec<String>,
    pub exclude_exts: Vec<String>,
    pub enable_gitignore: bool,
}
